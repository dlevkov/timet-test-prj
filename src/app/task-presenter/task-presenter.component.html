<mat-card>
  <p>{{ task.name }}</p>
  <p>{{ task.timer | async }}</p>
  <button mat-fab>
    <mat-icon>{{ task.buttonText }}</mat-icon>
  </button>
  {{ task | json }}
</mat-card>
