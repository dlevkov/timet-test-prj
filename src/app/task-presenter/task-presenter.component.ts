import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { TaskModel } from '../models/task-model';

@Component({
  selector: 'app-task-presenter',
  templateUrl: './task-presenter.component.html',
  styleUrls: ['./task-presenter.component.scss'],
})
export class TaskPresenterComponent {
  @Input() task: TaskModel;
  @Output() clicked = new EventEmitter<TaskModel>();

  public click() {
    this.clicked.emit(this.task);
  }
}
