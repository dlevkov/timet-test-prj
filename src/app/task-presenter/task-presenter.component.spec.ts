import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { TaskPresenterComponent } from './task-presenter.component';
import { MinuteSecondsPipe } from '../minute-seconds.pipe';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('TaskPresenterComponent', () => {
  let component: TaskPresenterComponent;
  let fixture: ComponentFixture<TaskPresenterComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TaskPresenterComponent, MinuteSecondsPipe],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TaskPresenterComponent);
    component = fixture.componentInstance;
    // fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
