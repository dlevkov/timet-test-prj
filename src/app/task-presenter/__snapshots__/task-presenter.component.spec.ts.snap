// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskPresenterComponent should render task 1`] = `
<div
  id="root1"
  ng-version="7.2.1"
>
  <mat-card
    class="row mat-card"
  >
    <p>
      some name
    </p>
    <p>
      10
    </p>
    <button
      mat-fab=""
    >
      <mat-icon
        aria-hidden="true"
        class="mat-icon material-icons mat-icon-no-color"
        role="img"
      >
        pause
      </mat-icon>
    </button>
  </mat-card>
</div>
`;
